extensions/wikidata/src/org/openrefine/wikidata/editing/EditBatchProcessor.java

```diff
@@ -182,6 +183,14 @@ public void performEdit()
                 // Existing entities
                 EntityUpdate entityUpdate = update.toEntityUpdate(currentDocs.get(update.getEntityId().getId()));
                 editor.editEntityDocument(entityUpdate, false, summary, tags);
+                if (entityUpdate instanceof FullMediaInfoUpdate) {
```

Comment:
In this method we are increasingly relying on `instanceof` checks, that is a sign we need a different architecture I think. The existing one worked as long as we were doing all of our editing through Wikidata-Toolkit, but this is no longer the case with this. I think this can stay for now but I would welcome proposals for other architectures.